server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/package.scala

```diff
@@ -0,0 +1,50 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap
+
+import eu.timepit.refined.api.{RefType, Validate}
+import org.apache.james.jmap.model.Id.Id
+import play.api.libs.json.{JsError, JsObject, JsString, JsSuccess, Reads, Writes}
+
+package object json {
+
+  implicit def writeRefined[T, P, F[_, _]](
```

Comment:
if it's a copy/paste of a lib, we should respect the license

Reply:
As @remk said it's copied from a lib with apache 2 license too. So is our license enough? should we put both? I'm not sure here

Reply:
Please add a comment: `code copied from https://xxxx/yyy`